rust/lanzatool/src/generation.rs

```diff
@@ -51,41 +47,45 @@ impl Generation {
             .try_into()
             .map_err(|err: &'static str| anyhow!(err))?;
 
-        let extensions = Self::extract_extensions(&bootspec)?;
-
         Ok(Self {
             version: link.version,
             specialisation_name: None,
-            spec: ExtendedBootJson {
-                bootspec,
-                extensions,
-            },
+            spec: ExtendedBootJson { bootspec },
         })
     }
 
-    fn extract_extensions(bootspec: &BootJson) -> Result<SecureBootExtension> {
-        Ok(Deserialize::deserialize(
-            bootspec.extensions.get("lanzaboote")
-            .context("Failed to extract Lanzaboote-specific extension from Bootspec, missing lanzaboote field in `extensions`")?
-            .clone()
-            .into_deserializer()
-        )?)
-    }
-
     pub fn specialise(&self, name: &SpecialisationName, bootspec: &BootJson) -> Result<Self> {
         Ok(Self {
             version: self.version,
             specialisation_name: Some(name.clone()),
             spec: ExtendedBootJson {
                 bootspec: bootspec.clone(),
-                extensions: Self::extract_extensions(bootspec)?,
             },
         })
     }
 
     pub fn is_specialized(&self) -> Option<SpecialisationName> {
         self.specialisation_name.clone()
     }
+
+    /// Describe the generation in a single line.
+    ///
+    /// This is currently implemented by poking around the filesystem to find the necessary data.
+    /// Ideally, the needed data should be included in the bootspec.
```

Comment:
Should this be feedback we give to the bootspec RFC?

Reply:
I spoke to @RaitoBezarius about this. I think he wants to address that at the next bootspec meeting.